Compilation errors for 64 bit iOS platforms (arm64, x86_64)

<!-- Please search existing issues for potential duplicates before filing yours:
https://github.com/godotengine/godot/issues?q=is%3Aissue
-->

**Godot version:**
Godot 3.1 
Branch: 3.1 
Commit hash: c4ff65207 
Commit name: "dist: Rename Linux mime type to follow system convention"


**OS/device including version:**
Any Device/Simulator iOS versions


**Issue description:**
Cannot compile under 64 bit iOS platforms (device, simulator)

terminal output:
Checking for C header file mntent.h... (cached) no
scons: done reading SConscript files.
scons: Building targets ...
[ 38%] Compiling ==> thirdparty/opus/opus_decoder.c
[ 38%] Compiling ==> thirdparty/opus/opus_encoder.c
[ 38%] Compiling ==> thirdparty/opus/opus_multistream_encoder.c
[ 38%] Compiling ==> thirdparty/opus/opus_multistream_decoder.c
[ 38%] Compiling ==> thirdparty/opus/repacketizer.c
[ 38%] Compiling ==> thirdparty/opus/analysis.c
[ 38%] Compiling ==> thirdparty/opus/mlp.c
[ 38%] Compiling ==> thirdparty/opus/mlp_data.c
[ 38%] Compiling ==> thirdparty/opus/info.c
In file included from thirdparty/opus/opus_decoder.c:41:
In file included from thirdparty/opus/celt/celt.h:41:
In file included from thirdparty/opus/celt/entenc.h:31:
In file included from thirdparty/opus/celt/entcode.h:35:
In file included from thirdparty/opus/celt/ecintrin.h:32:
In file included from thirdparty/opus/celt/arch.h:133:
thirdparty/opus/celt/arm/fixed_armv5e.h:124:20: error: unrecognized instruction mnemonic [12]
       "#MULT16_16\n\t"

Why fixed_armv5e.h included in x64/arm64 platforms? Probably issues in platform detection macroses

**Steps to reproduce:**
#build for x86_64 arch
scons p=iphone tools=no -j 10 target=release_debug arch=x86_64

#build for arm64 arch
scons p=iphone tools=no -j 10 target=release_debug arch=arm64


Build issue happen every time when building for 64bit ios arch

**Minimal reproduction project:**
Simply try to build godot for 64 bit ios platform

Building 3.1-stable causes same error on 64 bit archs for iOS

fixed_armv5e.h:124:20: error: unrecognized instruction mnemonic [12]
"#MULT16_16\n\t"

scons --version
SCons by Steven Knight et al.:
	script: v3.0.4.3a41ed6b288cee8d085373ad7fa02894e1903864, 2019-01-23 17:30:35, by bdeegan on kufra
	engine: v3.0.4.3a41ed6b288cee8d085373ad7fa02894e1903864, 2019-01-23 17:30:35, by bdeegan on kufra
	engine path: ['/usr/local/lib/scons/SCons']
Copyright (c) 2001 - 2019 The SCons Foundation

Build error still exist. 
Full build log attached in file:
[godot-build-log.txt](https://github.com/godotengine/godot/files/3042851/godot-build-log.txt)


[Initial build] /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -o thirdparty/opus/opus_decoder.iphone.opt.debug.arm64.o -c -DFIXED_POINT -DHAVE_CONFIG_H -DOPUS_ARM_OPT -fno-objc-arc -arch arm64 -fmessage-length=0 -fno-strict-aliasing -fdiagnostics-print-source-range-info -fdiagnostics-show-category=id -fdiagnostics-parseable-fixits -fpascal-strings -fblocks -fvisibility=hidden -MMD -MT dependencies -miphoneos-version-min=10.0 -isysroot /Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk -Wall -Werror=return-type -DZSTD_STATIC_LINKING_ONLY -w -DNDEBUG -DNS_BLOCK_ASSERTIONS=1 -O2 -ftree-vectorize -fomit-frame-pointer -DDEBUG_ENABLED -DNEED_LONG_INT -DLIBYUV_DISABLE_NEON -fno-exceptions -DGAME_CENTER_ENABLED -DSTOREKIT_ENABLED -DICLOUD_ENABLED -DIPHONE_ENABLED -DUNIX_ENABLED -DGLES_ENABLED -DCOREAUDIO_ENABLED -DMODULE_BMP_ENABLED -DMODULE_BULLET_ENABLED -DMODULE_CSG_ENABLED -DMODULE_DDS_ENABLED -DMODULE_ENET_ENABLED -DMODULE_FREETYPE_ENABLED -DMODULE_GDNATIVE_ENABLED -DMODULE_GDSCRIPT_ENABLED -DMODULE_GRIDMAP_ENABLED -DMODULE_HDR_ENABLED -DMODULE_JPG_ENABLED -DMODULE_MBEDTLS_ENABLED -DMODULE_MOBILE_VR_ENABLED -DMODULE_OGG_ENABLED -DMODULE_OPENSIMPLEX_ENABLED -DMODULE_OPUS_ENABLED -DPTRCALL_ENABLED -DGDSCRIPT_ENABLED -DMINIZIP_ENABLED -Ieditor -I. -I/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk/usr/include -I/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk/System/Library/Frameworks/OpenGLES.framework/Headers -I/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk/System/Library/Frameworks/AudioUnit.framework/Headers -Iplatform/iphone -Ithirdparty/zlib -Ithirdparty/zstd -Ithirdparty/libpng -Ithirdparty/libogg -Ithirdparty/opus -Ithirdparty/opus/celt -Ithirdparty/opus/opus -Ithirdparty/opus/silk -Ithirdparty/opus/silk/fixed -Ithirdparty/opus/silk/float thirdparty/opus/opus_decoder.c
In file included from thirdparty/opus/opus_decoder.c:41:
In file included from thirdparty/opus/celt/celt.h:41:
In file included from thirdparty/opus/celt/entenc.h:31:
In file included from thirdparty/opus/celt/entcode.h:35:
In file included from thirdparty/opus/celt/ecintrin.h:32:
In file included from thirdparty/opus/celt/arch.h:133:
thirdparty/opus/celt/arm/fixed_armv5e.h:124:20: error: unrecognized instruction mnemonic [12]
       "#MULT16_16\n\t"

--- a/modules/opus/SCsub
+++ b/modules/opus/SCsub
@@ -220,10 +220,15 @@ if env['builtin_opus']:
     ]
     env_opus.Append(CPPPATH=[thirdparty_dir + "/" + dir for dir in thirdparty_include_paths])
 
-    if env["platform"] == "android" or env["platform"] == "iphone":
-        if ("arch" in env and env["arch"] == "arm") or ("android_arch" in env and env["android_arch"] in ["armv6", "armv7"]):
+    if env["platform"] == "android":
+        if ("android_arch" in env and env["android_arch"] in ["armv6", "armv7"]):
             env_opus.Append(CFLAGS=["-DOPUS_ARM_OPT"])
-        elif ("arch" in env and env["arch"] == "arm64") or ("android_arch" in env and env["android_arch"] == "arm64v8"):
+        elif ("android_arch" in env and env["android_arch"] == "arm64v8"):
+            env_opus.Append(CFLAGS=["-DOPUS_ARM64_OPT"])
+    elif env["platform"] == "iphone":
+        if ("arch" in env and env["arch"] == "arm"):
+            env_opus.Append(CFLAGS=["-DOPUS_ARM_OPT"])
+        elif ("arch" in env and env["arch"] == "arm64"):
             env_opus.Append(CFLAGS=["-DOPUS_ARM64_OPT"])
 
     env_thirdparty = env_opus.Clone()

